encoding.bs

```diff
@@ -1125,6 +1108,60 @@ back to the caller that it has found a byte order mark and is therefore not usin
 encoding. The hook is to be invoked before <a>decode</a>, and it will return an encoding
 corresponding to the byte order mark found, or null otherwise.
 
+<hr>
+
+<p>To <dfn export>encode</dfn> an I/O queue of scalar values <var>ioQueue</var> given an encoding
+<var>encoding</var> and an optional I/O queue of bytes <var>output</var> (default « »), run these
+steps:
+
+<ol>
+ <li><p>Let <var>encoder</var> be the result of <a>getting an encoder</a> from <var>encoding</var>.
+
+ <li><p><a>Run</a> <var>encoder</var> with <var>ioQueue</var>, <var>output</var>, and
+ "<code>html</code>".
+
+ <li><p>Return <var>output</var>.
+</ol>
+
+<p class="note no-backref">This is a legacy hook for HTML forms. Layering <a>UTF-8 encode</a> on top
+is safe as it never triggers <a>errors</a>. [[HTML]]
+
+<hr>
+
+<p>To <dfn export lt="get an encoder|getting an encoder">get an encoder</dfn> from an
+<a for=/>encoding</a> <var>encoding</var>:
+
+<ol>
+ <li><p>Assert: <var>encoding</var> is not <a>replacement</a> or <a>UTF-16BE/LE</a>.
+
+ <li><p>Return <var>encoding</var>'s <a for=/>encoder</a>.
+</ol>
+
+<p>To <dfn export>encode or fail</dfn> an I/O queue of scalar values <var>ioQueue</var> given an
+<a for=/>encoder</a> <var>encoder</var> and an I/O queue of bytes <var>output</var>, run these
+steps:
+
+<ol>
+ <li><p>Let <var>potentialError</var> be the result of <a>running</a> <var>encoder</var> with
+ <var>ioQueue</var>, <var>output</var>, and "<code>fatal</code>".
+
+ <li><p>If <var>potentialError</var> is an <a>error</a>, then return <a>error</a>'s
+ <a>code point</a>'s <a for="code point">value</a>.
+
+ <li><p>Return null.
+</ol>
+
+<div class=note>
+ <p>This is a legacy hook for URLs. The caller will have to keep an <a for=/>encoder</a> alive as
+ the <a>ISO-2022-JP encoder</a> can be in two different states when returning an <a>error</a>. That
+ also means that if the caller emits bytes to encode the error in some way, these have to be in the
+ range 0x00 to 0x7F, inclusive, excluding 0x0E, 0x0F, 0x1B, 0x5C, and 0x7E. [[URL]]
```

Comment:
Please add more warning text along the lines of:

Notably, if upon returning an error the ISO-2022-JP encoder is in the Roman state, unlike in all other cases, if the caller inserts the byte 0x5C into the output, it will not decode into U+005C (\\) when the byte stream is decoded. For this reason, applications using the non-UTF-8 encoder facilities of this specification for purposes other than the intended ones (form submission and URL parsing in the Web Platform) ought to take care to prevent the use of the ISO-2022-JP encoder in combination with replacement schemes, such as those of JavaScript and CSS, that use U+005C (\\) as part of the replacement syntax.

Reply:
I guess this could even continue:

...or pass the replacement syntax through the encoder.

Also, it might make sense to add ("e.g. `\u2603`").

Hence:

Notably, if upon returning an error the ISO-2022-JP encoder is in the Roman state, unlike in all other cases, if the caller inserts the byte 0x5C into the output, it will not decode into U+005C (\) when the byte stream is decoded. For this reason, applications using the non-UTF-8 encoder facilities of this specification for purposes other than the intended ones (form submission and URL parsing in the Web Platform) ought to take care to prevent the use of the ISO-2022-JP encoder in combination with replacement schemes, such as those of JavaScript and CSS, that use U+005C (\) as part of the replacement syntax (e.g. `\u2603`) or make sure to pass the replacement syntax through the encoder (unlike in the URL parsing case).

Reply:
> Notably, if upon returning an error the ISO-2022-JP encoder **_instance_** is in the Roman state

Reply:
Done.